samples/apps/github-qna-webapp-react/src/components/GitHubRepoSelection.tsx

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import { Body1, Button, Input, Label, Spinner, Title3 } from '@fluentui/react-components';
+import { ArrowDownload16Regular, CheckmarkCircle20Filled, ErrorCircle20Regular } from '@fluentui/react-icons';
+import { FC, useEffect, useState } from 'react';
+import { useSemanticKernel } from '../hooks/useSemanticKernel';
+import { IKeyConfig } from '../model/KeyConfig';
+
+interface IData {
+    uri: string;
+    keyConfig: IKeyConfig;
+    prevProject: string;
+    prevBranch: string;
+    onBack: () => void;
+    onLoadProject: (project: string, branch: string) => void;
+}
+
+const enum DownloadState {
+    Setup = 0,
+    Loading = 1,
+    Loaded = 2,
+    Error = 3,
+}
+
+const GitHubProjectSelection: FC<IData> = ({ uri, keyConfig, prevProject, prevBranch, onLoadProject, onBack }) => {
+    const [project, setProject] = useState<string>(prevProject);
+    const [branch, setBranch] = useState<string>(prevBranch);
+    const [downloadState, setDownloadState] = useState<DownloadState>(DownloadState.Setup);
+    const sk = useSemanticKernel(uri);
+
+    const isSameProjectAndBranch = () => {
+        return project === prevProject && branch === prevBranch && project !== '' && branch !== '';
+    };
+
+    const download = async () => {
+        try {
+            setDownloadState(DownloadState.Loading);
+            var result = await sk.invokeAsync(
+                keyConfig,
+                {
+                    value: project || '',
+                    inputs: [
+                        { key: 'repositoryBranch', value: branch || '' },
+                        { key: 'searchPattern', value: '*.md' },
+                    ],
+                },
+                'GitHubSkill',
+                'SummarizeRepository',
+            );
+            setDownloadState(DownloadState.Loaded);
+            console.log(result);
+        } catch (e) {
+            setDownloadState(DownloadState.Error);
+            alert('Something went wrong.\n\nDetails:\n' + e);
+        }
+    };
+
+    useEffect(() => {
+        if (isSameProjectAndBranch()) {
+            setDownloadState(DownloadState.Loaded);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [project]);
+
+    useEffect(() => {
+        if (isSameProjectAndBranch()) {
+            setDownloadState(DownloadState.Loaded);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [branch]);
+
+    return (
+        <div style={{ paddingTop: 20, gap: 20, display: 'flex', flexDirection: 'column', alignItems: 'left' }}>
+            <Title3 style={{ alignItems: 'left' }}>Enter in the GitHub Project URL</Title3>
```

Comment:
"Enter in" sounds weird to me...

Recommend either "Enter the GitHub Project URL" or just "Enter GitHub Project URL"

Reply:
Agreed - made the heading consistent with the menu text:

<img width="422" alt="image" src="https://user-images.githubusercontent.com/146438/225936372-099bce25-8b6c-47c4-a953-a769c1d98284.png">